src/sentry/tasks/weekly_escalating_forecast.py

```diff
@@ -0,0 +1,81 @@
+from datetime import datetime
+from typing import Any, Dict, List, Tuple
+
+from django.db import transaction
+
+from sentry.issues.escalating_issues_alg import issue_spike
+from sentry.models import Group, GroupStatus
+from sentry.models.groupforecast import GroupForecast
+from sentry.models.groupsnooze import GroupSnooze
+from sentry.tasks.base import instrumented_task
+from sentry.utils.json import JSONData
+
+BATCH_SIZE = 200
+GroupForecastTuple = Tuple[Group, List[int]]
+
+
+def query_groups_past_counts(groups: List[Group]) -> JSONData:
+    return {}
+
+
+def get_forecast_per_group(
+    group_forecast_list: List[GroupForecastTuple],
+    archived_groups: List[Group],
+    response: JSONData,
+) -> None:
+    """
+    Modifies `group_forecast_list` to contain a list of forecasted values for each group.
+
+    `group_forecast_list`: List of GroupForecastTuple to be modified
+    `archived_groups`: List of archived groups to be forecasted
+    `archived_groups`: Snuba response for group event counts
+    """
+    data_index = 0
+    time = datetime.now()
+    for group in archived_groups:
+        group_data: Dict[str, List[Any]] = {"intervals": [], "data": []}
+        while (
+            data_index < len(response["data"])
+            and group.id == response["data"][data_index]["group_id"]
+        ):
+            # Flatten data into lists to be used in issue_spike
+            group_data["intervals"].append(response["data"][data_index]["hourBucket"])
+            group_data["data"].append(response["data"][data_index]["count()"])
+            data_index += 1
+
+        forecasts = issue_spike(group_data, time)
+        forecasts_list = [int(forecast["forecasted_value"]) for forecast in forecasts]
+        group_forecast_list.append((group, forecasts_list))
+
+
+@instrumented_task(
+    name="sentry.tasks.weekly_escalating_forecast.run_escalating_forecast",
+    queue="weekly_escalating_forecast",
+    max_retries=5,
+)  # type: ignore
+def run_escalating_forecast() -> None:
+    until_escalating_groups = [gs.group for gs in GroupSnooze.objects.filter(until_escalating=True)]
+    ignored_groups_ids = [
+        group["id"] for group in Group.objects.filter(status=GroupStatus.IGNORED).values()
+    ]
+    archived_groups = [group for group in until_escalating_groups if group.id in ignored_groups_ids]
+    if not archived_groups:
+        return
+    archived_groups.sort(key=lambda x: int(x.id), reverse=True)
+    response = query_groups_past_counts(archived_groups)
+
+    group_forecast_list: List[GroupForecastTuple] = []
+    get_forecast_per_group(group_forecast_list, archived_groups, response)
```

Comment:
Return the data instead of modifying it.

According to the [FP basics](https://github.com/readme/guides/functional-programming-basics), your function should return data, or another function.